neo4j/src/main/java/org/hibernate/ogm/datastore/neo4j/dialect/impl/BaseNeo4jEntityQueries.java

```diff
@@ -467,6 +467,7 @@ private static String initFindEntitiesQuery(EntityKeyMetadata entityKeyMetadata,
 		queryBuilder.append( ENTITY );
 		queryBuilder.append( ":" );
 		appendLabel( entityKeyMetadata, queryBuilder );
+		queryBuilder.append( ")" );
```

Comment:
Now that you mention it I think we missed the tests for the foreach method. It's been introduced for the mass indexer so it is only tested if we can run with the dialect.

I will create a JIRA for that